### PR TITLE
Remove system_rpc_server dependency for pigweed-app/esp32

### PR DESCRIPTION
**Pull request by szatmz**
#### Problem
system_rpc_server is only needed if hdlc_example::Start() is used to run the echo RPC service. We want to move to the custom RPC service in examples/common/pigweed/RpcService{.h, .cpp}

 #### Summary of Changes
- Removes references to system_rpc_server and hdlc_example in config/esp32
- Replaces hdlc_example::Start() with RpcService::Start running on a FreeRTOS task
- introduces FreeRTOS mutex to synchronize pw_log and pw_rpc responses, as per requirements of RpcService

### CODE REVIEW

File: examples/pigweed-app/esp32/main/component.mk [hunk 1]
```diff
@@ -23,16 +23,41 @@ COMPONENT_DEPENDS := chip
...
-                ../third_party/connectedhomeip/third_party/pigweed/repo/pw_span/public                      \
-                ../third_party/connectedhomeip/third_party/pigweed/repo/pw_polyfill/public                  \
-                ../third_party/connectedhomeip/third_party/pigweed/repo/pw_polyfill/standard_library_public \
-                ../third_party/connectedhomeip/third_party/pigweed/repo/pw_polyfill/public_overrides        \
-                ../third_party/connectedhomeip/third_party/pigweed/repo/pw_status/public                    \
-                ../third_party/connectedhomeip/third_party/pigweed/repo/pw_preprocessor/public              \
-                ../third_party/connectedhomeip/third_party/pigweed/repo/pw_rpc/system_server/public         \
-                ../third_party/connectedhomeip/examples/platform/esp32/pw_sys_io/public                     \
+COMPONENT_ADD_INCLUDEDIRS += ../third_party/connectedhomeip/third_party/pigweed/repo/pw_sys_io/public 				\
+							../third_party/connectedhomeip/third_party/pigweed/repo/pw_assert/public    			\
```
Comment (andy31415) [inline examples/pigweed-app/esp32/main/component.mk:29]: nit: should we align the \ here?

**Review by andy31415 — APPROVED** [review]

File: examples/pigweed-app/esp32/main/main.cpp [hunk 2]
```diff
@@ -16,18 +16,68 @@
...
 
     ESP_LOGI(TAG, "----------- chip-esp32-pigweed-example starting -----------");
-    hdlc_example::Start();
+
+    uart_mutex = xSemaphoreCreateMutex();
+    xTaskCreate(RunRpcService, "RPC", kRpcStackSizeBytes / sizeof(StackType_t), nullptr, kRpcTaskPriority, &rpcTaskHandle);
+
+    while (1)
+    {
+        vTaskDelay(50 / portTICK_PERIOD_MS);
```
Comment (mspang) [inline examples/pigweed-app/esp32/main/main.cpp:81]: Why does this wake every 500 ms?

**Review by mspang — APPROVED** [review]

File: examples/pigweed-app/esp32/main/component.mk [hunk 3]
```diff
@@ -23,16 +23,41 @@ COMPONENT_DEPENDS := chip
 
 COMPONENT_SRCDIRS :=                                                                \
   .                                                                                 \
+  ../../../common/pigweed															\
```
Comment (bzbarsky-apple) [inline examples/pigweed-app/esp32/main/component.mk:26]: Please align the backslashes.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@saurabhst ?

File: examples/pigweed-app/esp32/main/component.mk (hunk 3 above)
Comment (woody-apple) [reply examples/pigweed-app/esp32/main/component.mk:26]: @szatmz ?

**Review by msandstedt — APPROVED** [review]

**Pull request merged**